### DISCUSSION TITLE
provider_dependencies linting error triggered by generated provider from another file

### DISCUSSION THREAD

**Discussion by kennethjyang** [rrousselGit/riverpod#2824] [Q&A] [answered]
I am using a series of generated providers (defined with `@riverpod`) that utilize each other and are spread across a couple of files. For some reason, the linter says
> Generated providers should only depend on other generated providers. Failing to do so may break rules such as "provider_dependencies"

I looked into what the "[provider_dependencies](https://pub.dev/packages/riverpod_lint#provider_dependencies-riverpod_generator-only)" lint meant, but I am confused because I didn't use any scoped providers.

Here's my code:

**File: `view.dart`**
```dart
@riverpod
class View extends _$View {

  // [...]

  @override
  ViewState build(ViewProps props) {

    // Load block from database.
    return ref.watch(getDataByIdProvider(id: props.id)).when( // THIS PROVIDER IS HIGHLIGHTED BY THE LINTER
          // [...]
        );
  }
}
```

**File: `database.dart`**

```dart
@riverpod
Future<Isar> database(DatabaseRef ref) async {
  final dir = await getApplicationDocumentsDirectory();
  return Isar.open([BlockCollectionSchema], directory: dir.path);
}

@riverpod
Future<Data?> getDataById(getDataByIdRef ref,
    {required Id id}) {
  return ref.watch(databaseProvider).when( // NO LINTING ERROR HERE (?)
      data: (isar) {
        return isar.data.get(id);
      },
      error: (error, frame) => Future(() => null),
      loading: () => Future(() => null));
}
```

All of my providers build fine and work as expected. The chain of providers is `View <- getDataById <- database` where `<-` indicates one provider uses the following provider.

Why am I getting this lint error?

Thanks!

**Comment by AhmedLSayed9** [comment, marked as answer]
Sometimes the IDE just get confused. Re-running the build_runner command or restarting the IDE should solve it.

**Reply by kennethjyang** [reply]
wut.... I won't even question it 😆 Thanks, that worked.
